pkg/config/config.go

```diff
@@ -28,5 +57,58 @@ func NewConfig(configPath string) (*Config, error) {
 		config.Port = 3333
 	}
 
-	return config, nil
+	if servicePath == "" {
+		return nil, fmt.Errorf("Service Directory Missing")
+	}
+
+	if _, err := os.Stat(servicePath); os.IsNotExist(err) {
+		return nil, fmt.Errorf("%v: %s", err, servicePath)
+	}
+
+	files, err := ioutil.ReadDir(servicePath)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(files) == 0 {
+		return nil, fmt.Errorf("service directory is empty: %s", servicePath)
+	}
+
+	state := &State{
+		Config:   *config,
+		Services: map[string]Service{},
+	}
+
+	for _, file := range files {
+		serviceFilePath := filepath.Join(servicePath, file.Name())
+		bytes, err = ioutil.ReadFile(serviceFilePath)
+		if err != nil {
+			return nil, fmt.Errorf("Error opening config file: %v", err.Error())
+		}
+
+		serviceDef := ServiceDefinition{}
+		err = json.Unmarshal(bytes, &serviceDef)
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing service file: %s - %v", serviceFilePath, err.Error())
+		}
+
+		service := Service{
+			Name:        serviceDef.Name,
+			Description: serviceDef.Description,
+			DisplayName: serviceDef.DisplayName,
+			Tags:        serviceDef.Tags,
+			Plans:       make(map[string]Plan),
+		}
+
+		serviceID := uuidGenerator()
```

Comment:
commented about this above. we can just do `uuid.New` inside `NewState`

Reply:
This one was more for test reasons, but working on rewriting tests that we don't have to do that at all.
Us making UUIDs on the fly though have benefits and some problems. It means every time we start the broker all the services are new which would not work well with a platform. For the time being maybe it isn't important. Eventually I assume we are going towards a database which would help solve that issue.